pkg/object/object.go

```diff
@@ -0,0 +1,45 @@
+package object
+
+import (
+	"errors"
+
+	shell "github.com/ipfs/go-ipfs-api"
+	"github.com/sonr-io/sonr/pkg/schemas"
+)
+
+var (
+	ErrObjectNotUploaded = errors.New("error while uploading object")
+	ErrObjectEmpty       = errors.New("object cannot be empty")
+)
+
+type ObjectDefinition struct {
+	Did   string
+	Label string
+	Cid   string
+}
+
+/*
+	Object definition to be returned after object creation
+*/
+type ObjectUploadResult struct {
+	Code       int32
+	Definition *ObjectDefinition
+	Message    string
+}
+
+type AppObjectInternalImpl struct {
+	shell *shell.Shell
+}
+
+func New(schemaImpl schemas.AppSchemaInternal, shell *shell.Shell) AppObjectInternal {
```

Comment:
You're not using the value of `schemaImpl` here. You could keep this in `AppObjectInternalImpl` and then remove the need to pass it into the create method.

Reply:
that will work, but then each schema is 1-1 with an object.